lib/style-property-matcher.js

```diff
@@ -76,42 +49,34 @@ var StylePropertyMatcher = (function (window, document) { // jshint ignore:line
         return value.replace(re, "url($1)");
     };
 
-    /**
-     * Cross-browser getComputedStyle function.
+     /**
+     * There is a known issue in Safari browser:
+     * getComputedStyle(el, ":before") is empty if element is not visible.
      * 
-     * Known WebKit issue: 
-     * getComputedStyle(el, ":before").content returns empty string if element is not visible. 
-     */
-    var getComputedStyle = function (element, pseudoElement) {
-        var style;
-
-        if (isWebKit && pseudoElement) {
-            style = getPseudoElementComputedStyle(element, pseudoElement);
-        } else {
-            style = window.getComputedStyle(element, pseudoElement);
-        }
-        
-        return style;
-    };
-
-    /**
-     * Gets CSS property value
+     * To circumvent this issue we use getMatchedCSSRules instead.
+     *
+     * It appears that getMatchedCSSRules sorts the CSS rules
+     * in increasing order of specifities of corresponding selectors.
+     * We pick the css rule that is being applied to an element based on this assumption.
      * 
      * @param element       DOM node
      * @param pseudoElement Optional pseudoElement name
      * @param propertyName  CSS property name
      */
     var getComputedStylePropertyValue = function (element, pseudoElement, propertyName) {
-        var style = getComputedStyle(element, pseudoElement);
-        if (!style) {
-            return null;
+        var value = '';
```

Comment:
Previously I tried to minimize the number of modified functions. This time I did not care about it and modified the parent function directly. Also I've removed the check for quotes because it doesn't seem to be needed as every browser quotes content property values.